python/sbp/client/framer.py

```diff
@@ -39,17 +47,24 @@ class Framer(six.Iterator):
     def __init__(self,
                  read,
                  write,
+                 readinto=None,
                  verbose=False,
                  dispatcher=dispatch,
                  into_buffer=True,
-                 skip_metadata=False):
+                 skip_metadata=False,
+                 greedy=False):
         self._read = read
+        self._readinto = readinto
```

Comment:
How is this triggered?  I don't see a location where `readinto` is configured to be something other than `None`?

Reply:
Assuming you did something like this in piksi_tools:
```diff
diff --git a/piksi_tools/serial_link.py b/piksi_tools/serial_link.py
index c3fdc22f..6e32bb3b 100755
--- a/piksi_tools/serial_link.py
+++ b/piksi_tools/serial_link.py
@@ -338,7 +338,8 @@ def main(args):
     with Handler(Framer(driver.read,
                         driver.write,
                         args.verbose,
-                        skip_metadata=args.skip_metadata),
+                        skip_metadata=args.skip_metadata,
+                        readinto=True),
                  autostart=False) as link:
         # Logger with context
         with get_logger(args.log,
```

Reply:
```sh
pasi@pasi-linux:~/swiftnav/piksi_tools$ git stash pop
On branch silverjam/sbp2json
Your branch is up-to-date with 'origin/silverjam/sbp2json'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   piksi_tools/serial_link.py

no changes added to commit (use "git add" and/or "git commit -a")
Dropped refs/stash@{0} (a9698df8813e72a62c42b7948fc7bf20b5780ecb)
pasi@pasi-linux:~/swiftnav/piksi_tools$ git diff
diff --git a/piksi_tools/serial_link.py b/piksi_tools/serial_link.py
index c3fdc22..0c8ca8a 100755
--- a/piksi_tools/serial_link.py
+++ b/piksi_tools/serial_link.py
@@ -337,8 +337,10 @@ def main(args):
     driver = get_base_args_driver(args)
     with Handler(Framer(driver.read,
                         driver.write,
+                        driver.readinto,
                         args.verbose,
-                        skip_metadata=args.skip_metadata),
+                        skip_metadata=args.skip_metadata,
+                        greedy=True),
                  autostart=False) as link:
         # Logger with context
         with get_logger(args.log,
pasi@pasi-linux:~/swiftnav/piksi_tools$
```

Reply:
Currently `readinto` is implemented only in the `FileDriver`.

Reply:
👍